test/relations.test.js

```diff
@@ -918,6 +918,79 @@ describe('relations', function() {
       };
     });
 
+    describe('find over related model with options', function() {
+      after(function() {
+        Physician.clearObservers('access');
+        Patient.clearObservers('access');
+      });
+      before(function() {
+        Physician.observe('access', beforeAccessFn);
+        Patient.observe('access', beforeAccessFn);
+
+        function beforeAccessFn(ctx, next) {
+          ctx.query.where.realm = ctx.options.realm;
+          next();
+        }
+      });
+      it('should find be filtered from option', function(done) {
+        var id;
+        Physician.create(function(err, physician) {
+          physician.patients.create({name: 'a', realm: 'test'}, function(err, ch) {
```

Comment:
Unhandled `err`. Please add the following line as the first thing in the callback:

```js
if(err) return done(err);
```